infra/Pos/Binary/Infra/Slotting.hs

```diff
@@ -4,24 +4,22 @@ module Pos.Binary.Infra.Slotting
        (
        ) where
 
+import           Universum
+
 import           Data.Time.Units    (Millisecond)
 
-import           Pos.Binary.Class   (Cons (..), Field (..), deriveSimpleBi)
+import           Pos.Binary.Class   (Cons (..), Field (..), Bi (..), Size( VarSize ), deriveSimpleBi, getSize)
 import           Pos.Binary.Core    ()
 import           Pos.Core.Timestamp (TimeDiff)
-import           Pos.Core.Types     (EpochIndex)
-import           Pos.Slotting.Types (EpochSlottingData (..), SlottingData (..))
+import           Pos.Slotting.Types (EpochSlottingData (..), SlottingData, getSlottingDataMap, createSlottingDataUnsafe)
 
 deriveSimpleBi ''EpochSlottingData [
     Cons 'EpochSlottingData [
         Field [| esdSlotDuration :: Millisecond |],
         Field [| esdStartDiff    :: TimeDiff    |]
     ]]
 
--- CSL-1122: add a test for serialization of 'SlottingData'
-deriveSimpleBi ''SlottingData [
-    Cons 'SlottingData [
-        Field [| sdPenult      :: EpochSlottingData |],
-        Field [| sdLast        :: EpochSlottingData |],
-        Field [| sdPenultEpoch :: EpochIndex        |]
-    ]]
+instance Bi SlottingData where
+    size = VarSize $ \slottingData -> getSize $ getSlottingDataMap slottingData
+    put slottingData = put $ getSlottingDataMap slottingData
+    get = createSlottingDataUnsafe <$> get
```

Comment:
@ksaric it's true that our code can't create invalid `SlottingData`, but it's not true that nobody else can do it. In general you don't know where this data comes from. It can come from untrusted source.
And using `error` for deserialization errors leads to unpredictable behavior. Long ago we had segfaults because of it.